test/test_utilities.h

```diff
@@ -94,11 +94,14 @@ void Compare(const vector<Tensor>& v1, const vector<Tensor>& v2,
              float tolerance);
 
 // Compares two arguments
-template <class T>
+template <typename T>
 bool Compare(T arg0, T arg1) {
   return arg0 == arg1;
 }
 
+template <>
+bool Compare(float arg0, float arg1);
+
```

Comment:
Note to self: Fixing the `-O2` optimization error, that was ignoring the float specialization if the prototype is not specified in the header.